Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,460 +22,648 @@
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) {
+
+					const throw oip::UnhandledException(surface);
+
+					/* TO DO: Finish implementation of convertIfcSurfaceOfLinearExtrusion.
+
+					// Get swept curve and position.
+					EXPRESSReference<typename IfcEntityTypesT::IfcProfileDef> sweptSurfaceProfile = surface->SweptCurve.lock();
+					EXPRESSReference<typename IfcEntityTypesT::IfcAxis2Placement3D> sweptSurfacePlacement = nullptr;
+
+					if (surface->Position) {
+						EXPRESSReference<typename IfcEntityTypesT::IfcAxis2Placement3D> ref = surface->Position;
+						sweptSurfacePlacement = ref.lock();
+					}
+					else {
+						BLUE_LOG(warning) << "#" << swept_surface->getId() << " IfcSweptSurface without placement found.";
+					}
+
+					carve::math::Matrix surfaceMatrix(pos);
+					if (sweptSurfacePlacement) {
+						surfaceMatrix = pos * placementConverter->convertIfcAxis2Placement3D(sweptSurfacePlacement);
+					}
+
+
+					// Get extrude direction and depth.
+					EXPRESSReference<typename IfcEntityTypesT::IfcDirection>& linearExtrusionDirection = surface->ExtrudedDirection.lock();
+					double linearExtrusionDepth = surface->Depth;
+
+					*/
+
+				}
+
+				/*! \brief  Converts \c IfcSurfaceOfRevolution to ...
+				\param	surface	\c IfcSurfaceOfRevolution entity to be interpreted.
+				\param	pos
+				\param	polylineData
+				\note	The \c IfcSurfaceOfRevolution is a subtype of \c IfcSweptSurface.
+				*/
+
+				void convertIfcSurfaceOfRevolution(const EXPRESSReference<typename IfcEntityTypesT::IfcSurfaceOfRevolution>& surface,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) {
+
+					const throw oip::UnhandledException(surface);
+
+					/* TO DO: Finish implementation of convertIfcSurfaceOfRevolution.
+
+						// Get axis position.
+						EXPRESSReference<typename IfcEntityTypesT::IfcAxis1Placement>& axisPosition =
+							surface->AxisPosition.lock();
+
+					*/
+				}
+
+				//------------------------------------------------------------------------------------------------------------
+				// FaceConverter functions:
+				// convertIfcFaceList, convertIfcFace, convert3DPointsTo2D, triangulateFace, convertIfcCartesianPoint2DVector
+				//------------------------------------------------------------------------------------------------------------
+
+						/*! \brief  Converts \c IfcFace to a polygon and adds it to the carve PolyhedronData vector.
+						\param	faces \c IfcFace entity to be interpreted.
+						\param	pos
+						\param	itemData
+						\return polygon carve polygon
+						\note The \c IfcFaceList can be an open or closed shell.
+						*/
+
+				void convertIfcFaceList(const std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcFace>>& faces,
+					const carve::math::Matrix& pos,
+					EXPRESSReference<ItemData> itemData) {

```

Comment:
as said, this is std::shared_ptr, function is const throw(...); correct everywhere ...
```suggestion
					std::shared_ptr<ItemData> itemData) const throw(...) {
```

Reply:
Resolved: https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/a0cf345765ef7fda9e34e155f58d3058b5844210